### ISSUE TITLE
Error 503 during setup not recognized as such

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#1602]
![image](https://user-images.githubusercontent.com/6306961/54868387-db7d7e80-4d83-11e9-8b0b-b1f06dc67be0.png)

> Below are technical details about the error. This may be useful for debugging.<br><br><b>Error string</b>: Error: Api error response when POST-ing https://attester.flowcrypt.com/lookup/email string: email<br><br><b>Error stack</b>: Error: Api error response when POST-ing https://attester.flowcrypt.com/lookup/email string: email<br>    at Object.apiCall (chrome-extension://eaafkghcgcecpbpcbcdmfhgkaaaalbkn/js/common/api/api.js:719:11)<br><br>response:<br>[typeof:object:[object Object]] {"error":{"code":503,"message":"FlowCrypt Server temporarily unavailable, please try again\n\ntimeout expired","internal":null}}<b

It should have recognized it as a temporary server error (5xx)

**Comment by tomholub** [comment]
`isServerError` check is probably missing somewhere when this is being handled. Maybe inside `Api.err.eli5` or maybe somewhere in the `setup.ts` file.

**Comment by tomholub** [comment]
Same thing with 502:

![image](https://user-images.githubusercontent.com/6306961/57450274-9509b380-724d-11e9-990c-f3f1a4913f41.png)

From automated tests:

```
Page 1 (active) chrome-extension://nhinacaienkldfbmoljidbcefhfgablf/chrome/settings/setup.htm?acctEmail=flowcrypt.test.key.imported%40gmail.com

error: Failed to load resource: the server responded with a status of 502 (Bad Gateway)
request: 502 POST https://flowcrypt.com/attester/lookup/email
```

**Comment by michael-volynets** [comment]
There is a ServerError handler in `Api.err.eli5`.
Seems like it isn't an `AjaxError`, I see this code in `api.ts` after there is a result from ajax call:
```
    if (res && typeof res === 'object' && typeof (res as StandardErrorRes).error === 'object' && (res as StandardErrorRes).error.message) {
      throw new ApiErrorResponse(res as StandardErrorRes, req);
    }
```
Could it have happened after 500+ errors? What do you think?

**Comment by tomholub** [comment]
In the past, our servers used to send 4xx errors and 5xx errors as a http status 200 with error json in the body. In that case it would get to the code snippet you posted.

Recently I (cca a month ago?) changed the servers to return proper error codes, so a server error will return a 5xx status code. Which in JavaScript should be throwing the `AjaxError` (and never get to that line anymore).

When I looked at how we catch errors in general, it looked correct:
```js
isServerErr: (e: any): e is AjaxError => e instanceof AjaxError && e.status >= 500,
```
So you'll have to investigate why is it misbehaving during setup, and if that's a general problem throughout the extension, or just limited to the setup code.

**Comment by michael-volynets** [comment]
I have tested it and it works as it's written when the server returns 500, 
I see this page:
![image](https://user-images.githubusercontent.com/49104848/58249983-f4e17d80-7d68-11e9-967c-235891a26e20.png)

Maybe I don't understand what the result you want to see.

Here is a code from `Api.eli5`
```
} else if (Api.err.isServerErr(e)) {
        return 'Server responded with an unexpected error.';
      }
```

**Comment by tomholub** [comment]
Weird. Maybe it has since been fixed. Thanks for investigating this, let's close it.
